vm/src/builtins/type.rs

```diff
@@ -1337,28 +1355,25 @@ fn best_base(bases: &[PyTypeRef], vm: &VirtualMachine) -> PyResult<PyTypeRef> {
                 base_i.name()
             )));
         }
-        // candidate = solid_base(base_i);
-        // if (winner == NULL) {
-        //     winner = candidate;
-        //     base = base_i;
-        // }
-        // else if (PyType_IsSubtype(winner, candidate))
-        //     ;
-        // else if (PyType_IsSubtype(candidate, winner)) {
-        //     winner = candidate;
-        //     base = base_i;
-        // }
-        // else {
-        //     PyErr_SetString(
-        //         PyExc_TypeError,
-        //         "multiple bases have "
-        //         "instance lay-out conflict");
-        //     return NULL;
-        // }
+
+        let candidate = solid_base(base_i, vm);
+        if winner.is_none() {
+            winner = Some(candidate.clone());
+            base = Some(base_i.clone());
+        } else if winner.as_ref().unwrap().fast_issubclass(&candidate) {
+            // Do nothing
+        } else if candidate.fast_issubclass(winner.as_ref().unwrap()) {
+            winner = Some(candidate.clone());
+            base = Some(base_i.clone());
+        } else {
+            return Err(
+                vm.new_type_error("multiple bases have instance layout conflict".to_string())
+            );
+        }
     }
 
-    // FIXME: Ok(base.unwrap()) is expected
-    Ok(bases[0].clone())
+    debug_assert!(base.is_some());
+    Ok(base.unwrap())
```

Comment:
`unwrap` implies `panic!`. So `debug_assert!` is duplication.

```suggestion
    Ok(base.expect("best_base must exist"))
```